aries_cloudagent/resolver/base.py

```diff
@@ -38,6 +39,20 @@ class ResolverType(Enum):
     NON_NATIVE = "non-native"
 
 
+class ResolutionResult:
+    """Resolution Class to pack the DID Doc and the resolution information."""
+
+    def __init__(self, did_doc: DIDDocument, resolver_metadata: dict):
+        """Initialize Resolution.
+
+        Args:
+            did_doc: DID Document resolved
+            resolver_metadata: Resolving details
+        """
+        self.did_doc = did_doc
+        self.resolver_metadata = resolver_metadata
```

Comment:
I'd recommend going for just `metadata` to reduce redundancy a bit. `resolution_result.metadata` is a little more succinct than `resolution_result.resolver_metadata`.